addons/survey/controllers/main.py

```diff
@@ -361,7 +361,9 @@ def _prepare_question_html(self, survey_sudo, answer_sudo, **post):
                     'page_number': page_ids.index(survey_data['page'].id) + (1 if survey_sudo.progression_mode == 'number' else 0)
                 })
             elif survey_sudo.questions_layout == 'page_per_question':
-                page_ids = survey_sudo.question_ids.ids
+                page_ids = (answer_sudo.predefined_question_ids.ids
+                            if not survey_sudo.session_state
```

Comment:
I actually liked your previous test better, the one against the configuration of the survey?

Reply:
Well if it leads to crashes then we have to do something about it.

It is not perfect but it will do, you can go for your last solution and merge the commits.

Thank you for investigating 🙂